third_party/nvfuser/csrc/scheduler/registry.cpp

```diff
@@ -2038,10 +2044,272 @@ class PersistentKernelScheduler : public SchedulerEntry {
+          // we are interested only in mma ops with single output val
+          continue;
+        }
+        if (std::find(outputs.cbegin(), outputs.cend(), exprOutputs.front()) !=
+            outputs.cend()) {
+          mmaOutputs.push_back(expr);
+        }
+      }
+
+      mmaExprs = mmaOutputs;
+    }
+
+    if (mmaExprs.empty()) {
+      scheduler_debug_utils::canScheduleRejectReason(
+          ScheduleHeuristic::Matmul, "No output mma op to schedule");
+      return false;
+    }
+
+    // #3
+    {
+      // TODO: outline to ir_utils namespace
+      for (const auto* mmaExpr : mmaExprs) {
+        const auto& mmaExprInputs = mmaExpr->inputs();
+        if (mmaExprInputs.size() != 2) {
+          scheduler_debug_utils::canScheduleRejectReason(
+              ScheduleHeuristic::Matmul,
+              "One of mma op doesn't have 2 inputs, got: " +
+                  std::to_string(mmaExprInputs.size()));
+          return false;
+        }
+
+        // MMA inputs must be TVs
+        if (!mmaExprInputs[0]->isA<TensorView>() ||
+            !mmaExprInputs[1]->isA<TensorView>()) {
+          scheduler_debug_utils::canScheduleRejectReason(
+              ScheduleHeuristic::Matmul, "One of mma op inputs is not a TV");
+          return false;
+        }
+
+        const auto* inA = dynamic_cast<TensorView*>(mmaExprInputs[0]);
+        const auto* inB = dynamic_cast<TensorView*>(mmaExprInputs[1]);
+
+        // The number of IterDomains of MMA inputs must be the same
+        if (inA->nDims() != inB->nDims()) {
+          scheduler_debug_utils::canScheduleRejectReason(
+              ScheduleHeuristic::Matmul,
+              "Mma op inputs don't have the same number of IterDomains, 1st input(" +
+                  std::to_string(inA->nDims()) + "), 2nd input(" +
+                  std::to_string(inB->nDims()) + ")");
+          return false;
+        }
+
+        // The currently supported number of IterDomains perf MMA op input is 3
+        const size_t supportedDims = 3;
+        if (inA->nDims() != supportedDims) {
+          scheduler_debug_utils::canScheduleRejectReason(
+              ScheduleHeuristic::Matmul,
+              "Mma op inputs have unsupported number of IterDomains, got: " +
+                  std::to_string(inA->nDims()) + ", expected " +
+                  std::to_string(supportedDims));
+          return false;
+        }
+
+        // MMA input TVs must be in local memory
+        // NOTE: shouldn't buffers be in shared memory?
+        if (inA->getMemoryType() != MemoryType::Local ||
+            inB->getMemoryType() != MemoryType::Local) {
+          const auto memTypeToString =
+              [](const MemoryType memoryType) -> std::string {
+            switch (memoryType) {
+              case MemoryType::Global:
+                return "GLOBAL";
+              case MemoryType::Local:
+                return "LOCAL";
+              case MemoryType::Shared:
+                return "SHARED";
+            }
```

Comment:
nit: `MemoryType` can just be printed as there's `operator<<` overload for `MemoryType`.